components/Article/Progress/index.js

```diff
@@ -333,8 +321,19 @@ class Progress extends Component {
     }, 5000, { 'trailing': false })
 
     this.getMediaProgress = (mediaId) => {
-      const { mediaProgress } = this.state
-      return mediaProgress ? mediaProgress[mediaId] : undefined
+      return new Promise((resolve, reject) => {
```

Comment:
directly return `this.props.client.query().then()`, `this.props.client.query` already returns a promise, and `then` on a promise also always returns another promise.

No `catch` here, it's fine to swallow the error in the audio / video player but IMO no need to already swallow here. Also in the current implementation if `data.mediaProgress || data.mediaProgress === null` does not pass the promise is never resolved or rejected.

Reply:
API docs on `errors` is not super clear, at least I didn't find it. Therefore just ignore, I guess. In theory the promise should also be rejected.